clients/ui/bff/internal/mocks/model_catalog_settings_mock.go

```diff
@@ -0,0 +1,63 @@
+package mocks
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+
+	k8s "github.com/kubeflow/model-registry/ui/bff/internal/integrations/kubernetes"
+	"github.com/kubeflow/model-registry/ui/bff/internal/models"
+)
+
+type ModelCatalogSettingsRepositoryMock struct {
+}
+
+func NewModelCatalogSettingsRepository(logger *slog.Logger) (*ModelCatalogSettingsRepositoryMock, error) {
+	return &ModelCatalogSettingsRepositoryMock{}, nil
+}
+
+func (m *ModelCatalogSettingsRepositoryMock) GetAllCatalogSourceConfigs(_ context.Context, _ k8s.KubernetesClientInterface, _ string) (*models.CatalogSourceConfigList, error) {
+	allCatalogSourceConfigs := GetCatalogSourceConfigListMock()
+
+	return &allCatalogSourceConfigs, nil
+}
+
+func (m *ModelCatalogSettingsRepositoryMock) GetCatalogSourceConfig(_ context.Context, _ k8s.KubernetesClientInterface, _ string, catalogSourceId string) (*models.CatalogSourceConfig, error) {
+	catalogSourceConfig := CreateSampleCatalogSource(catalogSourceId, "catalog-source-1", "yaml")
+
+	return &catalogSourceConfig, nil
+}
+
+func (m *ModelCatalogSettingsRepositoryMock) CreateCatalogSourceConfig(_ context.Context, _ k8s.KubernetesClientInterface, _ string, sourceConfigPayload models.CatalogSourceConfigPayload) (*models.CatalogSourceConfig, error) {
+	var sourceName = sourceConfigPayload.Name
+	var sourceId = sourceConfigPayload.Id
+	var sourceType = sourceConfigPayload.Type
+
+	if sourceName == "" {
+		return nil, fmt.Errorf("source name is required")
+	}
+	if sourceId == "" {
+		return nil, fmt.Errorf("source ID is required")
+	}
+	if sourceType == "" {
+		return nil, fmt.Errorf("source type is required")
+	}
```

Comment:
It doesn't have to be part of this PR, but I wonder if we want to validate the required fields specific to each type. like for example a yaml-type source will require its yaml contents, and a huggingface-type source will require its allowedOrganization and apiKey. We could also bail if the type isn't one of those two types.

Reply:
I just added a few checks in here for mock, we will need to do the actual validation in `repositories/model_catalog_settings.go`